src/OrbitQt/CallTreeWidget.cpp

```diff
@@ -518,6 +567,19 @@ void CallTreeWidget::OnCustomContextMenuRequested(const QPoint& point) {
     for (const FunctionInfo* function : functions) {
       app_->ShowSourceCode(*function);
     }
+  } else if (action->text() == kActionSelectCallstacks) {
+    absl::flat_hash_set<orbit_client_data::CallstackEvent> selected_callstack_events =
+        GetCallstackEventsUnderSelection(selected_tree_indices);
+    ORBIT_CHECK(!selected_callstack_events.empty());
+    bool origin_is_multiple_threads =
+        std::any_of(selected_callstack_events.begin(), selected_callstack_events.end(),
+                    [first_callstack_event = *selected_callstack_events.begin()](
+                        const orbit_client_data::CallstackEvent& callstack_event) -> bool {
+                      return callstack_event.thread_id() != first_callstack_event.thread_id();
+                    });
+    app_->SelectCallstackEvents(
+        {selected_callstack_events.begin(), selected_callstack_events.end()},
```

Comment:
This is an undesired copy from `absl::flat_hash_set` to `std::vector`. One option is to make `OrbitApp::SelectCallstackEvents` a template, and another to just duplicate the method. But I don't like these ideas so please let me know what you think!

Reply:
Well I don't mind the extra copy here too much (especially as it is only temporary and only once on user action).

In general it would be great to have `SelectCallstackEvents` any kind of **iterable** collection of callstack events. But I see the burden of introducing templates. So I'd leave this to you (or the other reviewers).

One thing: If you decide to keep it as it, I would at least add a comment stating that the extra copy is intended.

Reply:
I assume the flat_hash_set is needed to avoid duplicate callstack events in the list, right?

I would also go with the copy! We have to clean up OrbitApp at some point. :innocent:

Reply:
Yes, that's the reason for `flat_hash_set`, the top-down view contains each `CallstackEvent` twice beacuse of the `(all threads)` tree.

I left the copy as it was and added a comment. Thanks!